src/ShalekKavy.Api/Controllers/BeveragesController.cs

```diff
@@ -0,0 +1,135 @@
+﻿using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
+using ShalekKavy.Api.Models.Enums;
+using ShalekKavy.Api.Services;
+using ShalekKavy.Api.Validation;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ShalekKavy.Api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class BeveragesController : ControllerBase
+    {
+        private readonly IBeverageRepository _repository;
+        public BeveragesController(IBeverageRepository repository)
+        {
+            _repository = repository;
+        }
+        // GET: BeverageController>
+        [HttpGet]
+        public async Task<IActionResult> GetBeverages()
+        {
+            var beverages = await _repository.GetAll();
+            return Ok(beverages);
+        }
+        // GET <BeverageController>/5
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetBeverageById(string id)
+        {
+            var beverage = await _repository.GetById(id);
+
+            if (beverage == null)
+            {
+                return BadRequest("A beverage with that Id does not exist.");
+            }
+
+            return Ok(beverage);
+        }
+        // GET <BeverageController>/name/{name}
+        [HttpGet("name/{name}")]
+        public async Task<IActionResult> GetByBeverageName(string name)
+        {
+            var beverage = await _repository.GetByBeverageName(name);
+
+            if (beverage == null)
+            {
+                return BadRequest("A beverage with that name does not exist.");
+            }
+
+            return Ok(beverage);
+        }
+        // GET <BeverageController>/type/{type}
+        [HttpGet("type/{type}")]
+        public async Task<IActionResult> GetByBeverageType(BeverageType type)
+        {
+            var beverages = await _repository.GetByBeverageType(type);
+
+            if (!beverages.Any())
+            {
+                return BadRequest("Beverages of that type do not exist.");
+            }
+
+            return Ok(beverages);
+        }
+        // POST <BeverageController>/add
+        [HttpPost("add")]
+        public async Task<IActionResult> AddBeverage([FromBody] Beverage beverage)
+        {
+            // VALIDATION - 1. Instantiate the validator object 
+            BeverageValidator validator = new BeverageValidator();
```

Comment:
Is there a way to use Dependency Injection to create singleton validators? You are not holding any state in those validators so I don't see a reason why you should create a new instance on every request.

Reply:
Yes, there is! I've updated the api validation to use DI